src/bsoncxx/test/bson_b_date.cpp

```diff
@@ -0,0 +1,22 @@
+#include "catch.hpp"
+
+#include <bsoncxx/types.hpp>
+#include <chrono>
+
+TEST_CASE("time_point is converted to b_date and back", "[bsoncxx::types::b_date]") {
+    using bsoncxx::types::b_date;
+    using std::chrono::system_clock;
+    using std::chrono::time_point_cast;
+    using std::chrono::milliseconds;
+
+    system_clock::time_point now1, now2;
+
+    now1 = system_clock::now();
+    b_date d{now1};
+    now2 = d;
+
+    //std::cout << "now1: " << time_point_cast<milliseconds>(now1).time_since_epoch().count() << std::endl;
+    //std::cout << "now2: " << time_point_cast<milliseconds>(now2).time_since_epoch().count() << std::endl;
+
+    REQUIRE(time_point_cast<milliseconds>(now1) == time_point_cast<milliseconds>(now2));
```

Comment:
can we also add a test that creates two b_date objects from the same time_point and then checks that their `value` members are equal?